cmd/kubeadm/app/cmd/phases/workflow/phase.go

```diff
@@ -0,0 +1,50 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package workflow
+
+// Phase provide an interface defining the common behavior of a phase of a kubeadm
+// composable workflows, thus decoupling the workflow execution logic from the concrete
+// phase implementation and allowing usage of custom phase types when necessary.
+type Phase interface {
```

Comment:
Too many `Get` prefixes in the methods of this interface, but that may be just me.

Reply:
This was intentional.

I think that the main usage of phases will be trough `SimplePhase` type instantiation
```
var myPhase = SimplePhase{
      Name:  "myPhase",
      Short: "A phase of a kubeadm composable workflow...",
      Run: func(ctx interface{}) error {
            ....
}
``` 
So I preserved names without suffix or prelix like e.g. `Name` for that struct

Instead, I think that the `Phase` interfaces (where I was forced to add `Get` prefix for avoiding name conflicts with ˆˆˆ)  will be used only if we need custom phase types along the way.

Reply:
a semantical problem is that the getters (e.g. GetName) are acting on fields that are already public (e.g. Name).
the idea of a getter would be to make a private field public. (name -> Name())

do we really need the interface to be public? perhaps we can make it private to solve the getter naming issue.
we can potentially rename SimplePhase -> Phase and Phase can be just a struct like cobra Command.
then new phase types can use composition.

https://play.golang.org/p/kI7xZWpH6Xv

something else to note is that we don't really need the interface if all phases (consumers) are just going to create SimplePhase objects, instead of inheriting from SimplePhase.

Reply:
We can talk about semantics for weeks 😉, but honestly I don't see this as a issue.

IMO the most important things is to getting phase moving, and we can eventually iterate on this detail afterwards

> something else to note is that we don't really need the interface if all phases 

I'm not sure about this, because might be we decide to implement a custom case for generating certs (accepting cert name as input) instead of generating ~10 simple phases.
This is why I'm keeping an interface in the picture